src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,97 @@
+import Issue from "../../issueHandling/issue";
+import { concatOneIssueList } from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import { UnexpectedError } from "./Errors";
+import modes from "./predicateList.json";
+
+export type Mode = "propositionalLogic" | "firstOrderPredicateLogic" | "test";
+
+/**
+ * Since the user input most likely lacks the correct predicate,
+ * it needs to be added. For easier maintainability in case
+ * of an expanding predicate list, the corresponding predicate is
+ * added with respect to the current @param mode. This can, for
+ * example correspond to the concept of "Spielwiese". The user should
+ * be able to just stay inside a mode and not have to add this mode
+ * himself. Hence, the mode needs to be read out from the browser.
+ */
+export function addPredicate(userInput: string, mode: Mode) {
+	for (const predicate of modes.predicates) {
+		if (predicate === mode) {
+			return getPrologPredicateForMode(mode).concat("(").concat(userInput).concat(").");
+		}
+	}
+	return "Es gibt kein passendes Prädikat.";
+}
+
+export function getPrologPredicateForMode(mode: Mode) {
+	switch (mode) {
+		case "propositionalLogic": {
+			return "diproche";
+		}
+		case "firstOrderPredicateLogic": {
+			return "diproche_fo";
+		}
+		case "test": {
+			return "teste";
+		}
+		default: {
+			throw(UnexpectedError);
+		}
+	}
+}
+
+// Assuming "wrongWord" is not an allowed word, then a
+// vocaberror is an error like "the word "wrongWord"
+// in the text "This text contains a wrongWord".
+export function getVocabErrors(userInput: string): Issue[] | void {
```

Comment:
When is the return type a Issue[] and when void?

Reply:
That was a leftover from some debugging I did before a commit. Good thing you catched it. void doesn't serve any purpose here.